store/store.go

```diff
@@ -30,7 +31,9 @@ type GetRequest struct {
 var ErrNotExists = errors.New("Requested object does not exist")
 
 type Store interface {
-	Store(ctx context.Context, obj []byte) (string, error)
+	StoreBytes(ctx context.Context, obj []byte) (string, error)
```

Comment:
By default I would only have `Store`, and would make a `StoreBytes` helper somewhere that wraps the bytes in a `bytes.Buffer`. 

If an implementation wants to do something more efficient when it knows it has a byte buffer, it can do a type assertion to `*bytes.Buffer` to check for that.